entry_test.go

```diff
@@ -43,17 +50,234 @@ func TestEntryPanicln(t *testing.T) {
 		case *Entry:
 			assert.Equal(t, "kaboom", pVal.Message)
 			assert.Equal(t, errBoom, pVal.Data["err"])
+			assert.Regexp(t, `^level=panic msg=kaboom err="boom time"`, outBuffer.String())
 		default:
 			t.Fatalf("want type *Entry, got %T: %#v", pVal, pVal)
 		}
 	}()
 
-	logger := New()
-	logger.Out = &bytes.Buffer{}
+	logger.Out = &outBuffer
 	entry := NewEntry(logger)
 	entry.WithField("err", errBoom).Panicln("kaboom")
 }
 
+// TestEntryFormattingError tests proper handling of any field formatting errors
+func TestEntryFormattingError(t *testing.T) {
+	logger := New()
+	// Turning off timestamp and color marks parsing the result easier:
+	logger.Formatter = &TextFormatter{
+		DisableTimestamp: true,
+		DisableColors:    true,
+	}
+	// Regular Expressions used later
+	keyValPartsPattern := regexp.MustCompile(`([^\s"=]*|"(?:[^\\"]|\\.)*")=([^\s"=]*|"(?:[^\\"]|\\.)*")(?:\s|$)`)
```

Comment:
with a json formatter we won't need a complicated regexp to parse the ouput of a trace

Reply:
Agreed